cluster.c

```diff
@@ -597,18 +699,34 @@ RRStatus ShardingInfoValidateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
 
 RRStatus ShardingInfoUpdateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
 {
-    ShardingInfo *si = rr->sharding_info;
+    if (!strncmp(rr->snapshot_info.dbid, new_sg->id, sizeof(new_sg->id))) {
+        /* TODO: This is our cluster, so will have to handle add and removing nodes */
+    } else {
+        ShardGroup *sg = getShardGroupById(rr, new_sg->id);
+        if (sg == NULL) {
+            return RR_ERROR;
+        }
 
-    if (new_sg->id < 1 || new_sg->id > si->shard_groups_num)
-        return RR_ERROR;
+        sg->nodes_num = new_sg->nodes_num;
+        sg->nodes = RedisModule_Realloc(sg->nodes, sizeof(ShardGroupNode) * sg->nodes_num);
+        memcpy(sg->nodes, new_sg->nodes, sizeof(ShardGroupNode) * sg->nodes_num);
+    }
+
+    return RR_OK;
+}
 
-    ShardGroup *sg = si->shard_groups[new_sg->id - 1];
+ShardGroup * getShardGroupById(RedisRaftCtx *rr, char id[RAFT_DBID_LEN+1])
```

Comment:
Probably better to take `const char *id` and if we need a null-terminated version, make a local copy on the stack, no?

Reply:
note quite sure I understand

Reply:
changed it to char *, unsure there would ever be a need to copy, from what I can tell, its always a char * c string (hence the +1)